integration/generic-tests/kexec_test.go

```diff
@@ -144,3 +146,34 @@ func TestMountKexecLoadCustomDTB(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestKexecLinuxImageCfgFile(t *testing.T) {
+	// TODO: support arm
+	if vmtest.TestArch() != "amd64" && vmtest.TestArch() != "arm64" {
+		t.Skipf("Test not supported on %s", vmtest.TestArch())
+	}
+
+	cfg := []byte("{ \"InitrdPath\": \"/testdata/initramfs.cpio\", \"KernelPath\": \"/testdata/kernel\", \"Cmdline\": \"/proc/cmdline\", \"Name\": \"testloadconfig\" }")
+
+	if err := os.WriteFile(filepath.Join(t.TempDir(), "linux_image_cfg.json"), cfg, 0777); err != nil {
+		t.Fatalf("Failed to setup test cfg file: %v", err)
+	}
+
+	q, cleanup := vmtest.QEMUTest(t, &vmtest.Options{
+		TestCmds: []string{
+			"echo kexecloadresult ?(kexec -d -l -I /testdata/linux_image_cfg.json)",
+		},
+		TmpDir: t.TempDir(),
```

Comment:
"Each subsequent call to t.TempDir returns a unique directory"

this won't work, from the docs

Reply:
G.... I thought same per test function life 

done.